### ISSUE TITLE
Find File Exception

### ISSUE THREAD

**Issue by n2o** [zmsp/sonar-clojure#3]
I have SonarQube running in a docker container and want to analyze / visualize my Clojure code. When I add your jar file (manually compiled with `mvn package`) to my extensions, I get the following errors when calling `sonar-runner` on my local machine:

```
Clojure project detected, running sonar-clojure
●▬▬▬▬▬▬▬▬▬● Running file finder ●▬▬▬▬▬▬▬▬▬●
ERROR: ▂▃▅▇█▓▒░ FIND FILE: EXCEPTION  ░▒▓█▇▅▃▂
java.lang.IllegalArgumentException: Parameter 'directory' is not a directory
	at org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:545)
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:521)
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:691)
	at org.sonar.plugins.clojure.InspectClojureSensor.findCLJFile(InspectClojureSensor.java:215)
	at org.sonar.plugins.clojure.InspectClojureSensor.buildFileProperties(InspectClojureSensor.java:191)
	at org.sonar.plugins.clojure.InspectClojureSensor.analyse(InspectClojureSensor.java:98)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:175)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:262)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:257)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:247)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)

▂▃▅▇█▓▒░Find Files Exception░▒▓█▇▅▃▂
java.lang.IllegalArgumentException: Parameter 'directory' is not a directory
	at org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:545)
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:521)
	at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:691)
	at org.sonar.plugins.clojure.InspectClojureSensor.findCLJFile(InspectClojureSensor.java:215)
	at org.sonar.plugins.clojure.InspectClojureSensor.buildFileProperties(InspectClojureSensor.java:191)
	at org.sonar.plugins.clojure.InspectClojureSensor.analyse(InspectClojureSensor.java:98)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:175)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:262)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:257)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:247)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
▂▃▅▇█▓▒░Building file properties error░▒▓█▇▅▃▂
java.lang.NullPointerException
	at org.sonar.plugins.clojure.InspectClojureSensor.buildFileProperties(InspectClojureSensor.java:191)
	at org.sonar.plugins.clojure.InspectClojureSensor.analyse(InspectClojureSensor.java:98)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:57)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:49)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:175)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:262)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:257)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:247)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:143)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:128)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:123)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:77)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
●▬▬▬▬▬▬▬▬▬● Running Eastwood ●▬▬▬▬▬▬▬▬▬●
```

I have kibit and eastwood running in my project. And it is not the only project for my SonarQube instance. Other projects work as expected (in Java, Python and JavaScript), but my Clojure project does not work with your plugin. It throws `java.lang.IllegalArgumentException: Parameter 'directory' is not a directory`.

My system:
```
INFO: SonarQube Scanner 3.0.1.733
INFO: SonarQube server 6.3.1
INFO: Java 1.8.0_131 Oracle Corporation (64-bit)
INFO: Linux 4.10.13-1-ARCH amd64
```

My sonar-project.properties:
```
# must be unique in a given SonarQube instance
sonar.projectKey=foo
# this is the name and version displayed in the SonarQube UI. Was mandatory prior to SonarQube 6.1.
sonar.projectName=foo
sonar.projectVersion=0.1
 
# Path is relative to the sonar-project.properties file. Replace "\" by "/" on Windows.
# This property is optional if sonar.modules is set. 
sonar.sources=.
 
# Encoding of the source code. Default is default system encoding
#sonar.sourceEncoding=UTF-8

sonar.import_unknown_files=true

sonar.exclusions=node_modules/**, docs/**, target/**, **/*.min.*, **/bower_components/**, **/*.js
```

In fact, it's a ClojureScript application, which explains the exclusions. But kibit and eastwood are cljs-compliant.

**Comment by zmsp** [comment]
Thank you for the issue. I will look into it this weekend.

**Comment by zmsp** [comment]
It should be fixed now on c072f2d81ec792c44fa6bc6654ea2b2a77ab57ca
See release for jar. 

Also fyi, sonar.sources is something required by sonar-scanner but this program isn't really concerned about the value. The program just looks at base directory of the project.

Thanks.

**Comment by n2o** [comment]
This problem is fixed now, thanks. But another occured. I'll open a new issue for this.
Thank you very much!
